src/emd/rt_bse.F

```diff
@@ -0,0 +1,1420 @@
+         ! Next step is to convert the greens full matrix to dbcsr matrix
+         CALL dbt_copy_matrix_to_tensor(greens_dbcsr, rtbse_env%greens_dbt)
+         ! Then contract it
+         ! no scaling applied - this has to be applied externally
+         CALL dbt_contract(alpha=1.0_dp, &
+                           tensor_1=rtbse_env%t_3c_work_RI_AO__AO, &
+                           tensor_2=rtbse_env%greens_dbt, &
+                           beta=0.0_dp, &
+                           tensor_3=rtbse_env%t_3c_work2_RI_AO__AO, &
+                           contract_1=[2], notcontract_1=[1, 3], map_1=[1, 3], &
+                           contract_2=[2], notcontract_2=[1], map_2=[2])
+         ! workspace 2 now contains C ^ P _ (mu beta) sum _ nu B ^ P _ (nu beta) g _ (mu nu)
+         CALL dbt_contract(alpha=prefactor, &
+                           tensor_1=rtbse_env%t_3c_w, &
+                           tensor_2=rtbse_env%t_3c_work2_RI_AO__AO, &
+                           beta=0.0_dp, &
+                           tensor_3=rtbse_env%sigma_dbt, &
+                           contract_1=[1, 3], notcontract_1=[2], map_1=[1], &
+                           contract_2=[1, 2], notcontract_2=[3], map_2=[2])!,&
+         !filter_eps=bs_env%eps_filter)
+         ! Finally, convert the COH tensor to matrix and then to fm matrix
+         CALL dbcsr_create(sigma_dbcsr, name="sigma", template=rtbse_env%rho_dbcsr)
+         CALL dbt_copy_tensor_to_matrix(rtbse_env%sigma_dbt, sigma_dbcsr)
+         CALL copy_dbcsr_to_fm(sigma_dbcsr, sigma_fm%matrix)
+         CALL dbcsr_release(sigma_dbcsr)
+         ! Clear workspaces - saves memory?
+         CALL dbt_clear(rtbse_env%t_3c_work_RI_AO__AO)
+         CALL dbt_clear(rtbse_env%t_3c_work2_RI_AO__AO)
+         CALL dbt_clear(rtbse_env%sigma_dbt)
+         CALL dbt_clear(rtbse_env%greens_dbt)
+         CALL timestop(handle)
+
+      END SUBROUTINE get_sigma_dbcsr
+! **************************************************************************************************
+!> \brief Creates the RI matrix and populates it with correct values
+!> \note Tensor contains Hartree elements in the auxiliary basis
+!> \param qs_env Quickstep environment - entry point of calculation
+!> \author Stepan Marek
+!> \date 01.2024
+! **************************************************************************************************
+      SUBROUTINE init_hartree(rtbse_env, v_dbcsr)
+         TYPE(rtbse_env_type), POINTER, INTENT(IN)          :: rtbse_env
+         TYPE(dbcsr_type), POINTER                          :: v_dbcsr
+         TYPE(post_scf_bandstructure_type), POINTER         :: bs_env
+         TYPE(libint_potential_type)                        :: coulomb_op
+         TYPE(cp_fm_type)                                   :: V_fm
+         TYPE(cp_fm_type)                                   :: metric_fm
+         ! TYPE(cp_fm_type), DIMENSION(:,:), ALLOCATABLE      :: metric_fm
```

Comment:
Do you still need this?

Reply:
No, you are completely correct - I deleted the commented declaration